util/logger/src/structured_logger.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright 2018 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use parking_lot::Mutex;
+use std::sync::mpsc::{channel, Receiver, Sender};
+use std::sync::Arc;
+
+use serde::Serialize;
+use serde_derive::Serialize;
+use serde_json;
+use serde_json::to_value;
+
+#[derive(Clone)]
+pub struct StructuredLogger {
+    // Wrap sender with mutex to get Sync trait
+    // To use in global.
+    sender: Arc<Mutex<Sender<serde_json::Value>>>,
+    receiver: Arc<Mutex<Receiver<serde_json::Value>>>,
+    enabled: Arc<Mutex<bool>>,
+}
+
+#[derive(Serialize)]
+pub struct Log {
+    pub level: String,
+    pub target: String,
+    pub message: String,
+    pub timestamp: String,
+}
+
+impl StructuredLogger {
+    pub fn create() -> StructuredLogger {
+        let (sender, receiver) = channel();
+        StructuredLogger {
+            sender: Arc::new(Mutex::new(sender)),
+            receiver: Arc::new(Mutex::new(receiver)),
+            enabled: Arc::new(Mutex::new(false)),
```

Comment:
How about use AtomicBool?

Reply:
I don't know well about atomic ordering.
miner's sealing_enabled always use Ordering::Relaxed.
Is it OK to use only Ordering::Relaxed to `enabled`?

Reply:
Use `SeqCst`, if it doesn't harm the performance.